staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -701,6 +705,10 @@ func (r *crdHandler) getOrCreateServingInfoFor(uid types.UID, name string) (*crd
 	// Create replicasPathInCustomResource
 	replicasPathInCustomResource := managedfields.ResourcePathMappings{}
 	for _, v := range crd.Spec.Versions {
+		if !v.Served {
+			continue
+		}
```

Comment:
I think we should also not short-circuit here... managed field info for all versions is needed even when serving for different versions